### DISCUSSION TITLE
Extracting table does not use correct borders

### DISCUSSION THREAD

**Discussion by tz850** [jsvine/pdfplumber#1244] [Ask for help with specific PDFs]
## Describe the bug

When extracting the table content on the page, it does not follow the table borders, but seems to use the label of the chart below.
This problem causes the text extracted from the cell to be incorrect. For example, in the third and fourth columns of the first row, the correct text should be "
![txt1](https://github.com/user-attachments/assets/7c26568b-e4b9-43ad-a26a-a6c96d2a6bd2)
", 
but the actual extract_tables function‘s result is "
![txt2](https://github.com/user-attachments/assets/7b952de8-5c08-4b7b-af16-31fc5e35aee9)
"

This is the original page.
![image](https://github.com/user-attachments/assets/12228f81-3c4f-44e5-ba97-96b1222a0946)

Here is an image of the debug_tablefinder output. The cell borders pointed to by the arrows in the figure are not the actual borders of the table.
![tableimage](https://github.com/user-attachments/assets/c6f2cc16-fa42-4368-a617-a57bee363110)

## Have you tried [repairing](https://github.com/jsvine/pdfplumber/blob/stable/docs/repairing.md) the PDF?

I run ghostscript directly to output the repaired pdf file, and the problem is the same.

## Code to reproduce the problem
```
import pdfplumber
from pdfminer.high_level import extract_pages
from pdfminer.layout import LTTextContainer, LTTextLine
from PIL import Image, ImageDraw, ImageFont
import matplotlib.pyplot as plt

with pdfplumber.open("f://repaired.pdf") as pdf:
    pages_count = len(pdf.pages)    
    print(pages_count)

    page = pdf.pages[0]

    # This code saves the debug visual output.
    im = page.to_image(resolution=200)
    im.reset().debug_tablefinder()
    im.save("tableimage.png", format="PNG")

    table_obj_array = page.find_tables()
    table_content_array = page.extract_tables()

```

## PDF file
[sample.pdf](https://github.com/user-attachments/files/18279681/sample.pdf)

## Expected behavior

I hope that the division of table cells will not be affected by other tables or charts.

## Actual behavior

*What actually happened, instead?*

## Environment

- pdfplumber version: [0.11.4]
- Python version: [3.10.0]
- OS: [windows 10]

## Additional context

*Add any other context/notes about the problem here.*

**Comment by jsvine** [comment]
Hi @tz850, and thanks for providing the PDF and visual debugging output. This is an interesting edge case, where the page has a lot of other graphical objects (`len(page.lines) == 635` and `len(page.curves) == 65`). In this case, they're shifting the ultimate placement of the vertical lines in the table due to the `snap_tolerance = 3` default — a setting that usually *helps* with proper table detection but here it does not. The good news is that you can adjust this:

```python
im.reset().debug_tablefinder({
    "snap_tolerance": 0,
})
```

... seems to get you want you'd want:

![image](https://github.com/user-attachments/assets/ce9e1b8b-9a67-43c7-82ff-165a466b786a)

**Reply by tz850** [reply]
Thanks for your reply, this solved my problem. Happy New Year!
